sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/transforms/Count.java

```diff
@@ -106,30 +118,68 @@ public void processElement(ProcessContext c) {
   /**
    * A {@link CombineFn} that counts elements.
    */
-  private static class CountFn<T> extends CombineFn<T, Long, Long> {
+  private static class CountFn<T> extends CombineFn<T, long[], Long> {
```

Comment:
Worth a comment explaining that the accumulator is always size 1, and is used as a mutable box for the long.